static/js/composebox_typeahead.js

```diff
@@ -349,9 +349,18 @@ exports.content_typeahead_selected = function (item) {
                 + '#**' + item.name + '** ');
         $(document).trigger('streamname_completed.zulip', {stream: item});
     } else if (this.completing === 'syntax') {
-        rest = "\n" + beginning.substring(beginning.length - this.token.length - 4,
-                beginning.length - this.token.length).trim() + rest;
-        beginning = beginning.substring(0, beginning.length - this.token.length) + item + "\n";
+        if (rest === '') {
+            // If cursor is at end of input ("rest" is empty), then
+            // complete the token before the cursor, and add a closing fence
+            // after the cursor
+            rest = "\n" + beginning.substring(beginning.length - this.token.length - 4,
```

Comment:
I think it makes sense to have `rest` before `beginning`, since `rest` depends on the original value of `beginning`. But I just isolated out another local variable, which seems to make this a bit cleaner.